### PR TITLE
在 SecretFlow 中实现个性化联邦策略 FedPer #995

### PR DESCRIPTION
**Pull request by Starlight039**
#995 #1066 
# Preamble

For Pull Request initiator, please read the following steps carefully. 请合并请求的发起人仔细阅读下列步骤。

---

If you are creating a `Pull Request` about listed topic below, please go the the `Preview` tab and click the relative link:
如果您要创建一个有关下列内容的的合并请求，请切换到 `Preview` 面板并点击下方相关的链接:

* Select [Awesome-PETs](?expand=1&template=pull_request_template_pets.md) for files under docs/awesome-pets

---

Select a proper `Pull Request` template can accelarate the review process, thanks for your cooperation.
选择一个正确的`合并请求`模版可以加速审查流程，谢谢您的配合

If your `Pull Request` topic is not amoung the list above, please just delete all prepopulated template content.
如果您的`合并请求`内容不在上述的列表里，请删除提前填入的模版内容。

---

Thanks for your contribution to SecretFlow project
感谢您对于 SecretFlow 项目的贡献

### CODE REVIEW

**Comment by zhangxingmeng** [comment]
还需要增加pytest单测，可以参考下[fed_avg_g的单测](https://github.com/secretflow/secretflow/blob/main/tests/ml/nn/fl/strategy/test_fed_avg_tf.py) 或者https://github.com/secretflow/secretflow/blob/main/tests/ml/nn/fl/test_fl_model_tf.py

File: tests/ml/nn/fl/strategy/test_fed_per_torch.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        # Prepare dataset
+        x_test = torch.rand(128, 1, 28, 28)  # Randomly generated data
+        y_test = torch.randint(0, 10, (128,))  # Randomly generated labels for a 10-class task
+        test_loader = DataLoader(TensorDataset(x_test, y_test), batch_size=32, shuffle=True)
+        fed_per_worker.train_set = iter(test_loader)
+        fed_per_worker.train_iter = iter(fed_per_worker.train_set)
+
+        # Perform a training step
+        gradients = None
+        gradients, num_sample = fed_per_worker.train_step(
```
Comment (zhaocaibei123) [inline tests/ml/nn/fl/strategy/test_fed_per_torch.py:40]: 这里 gradients 应该是 weights 吧

**Review by zhaocaibei123 — APPROVED** [review]

**Comment by zhaocaibei123** [comment]
代码用black format下

**Review by zhaocaibei123 — APPROVED** [review]

**Review by ian-huu — DISMISSED** [review]
LGTM
